web/src/components/MetricRatio.tsx

```diff
@@ -1,6 +1,15 @@
-export function MetricRatio({ value, total, format }) {
-  const formattedValue = Number.isFinite(value) ? format(value) : '?';
-  const formattedTotal = Number.isFinite(total) ? format(total) : '?';
+import type { Maybe } from 'types';
+
+interface MetricRatioProps {
+  value: number;
+  total: Maybe<number>;
```

Comment:
While it's an improvement shouldn't this always be either null or a number? Would it not be better to mark the it as such instead of maybe number?

Reply:
Since `total` can be `capacity` which comes all the way back from `ZoneDetails`, it should either accept the same type or we should add a check somewhere in between that ensures it's only `null` and not `undefined`. I don't think there's much value in enforcing this right now though, so leaving it as `Maybe<number>` and then we can rethink the whole `undefined` vs `null` (vs `NaN`) later on :)

Reply:
We could add it as `total?: number | null` but I suppose that just does the same as `Maybe<number>`?

Reply:
`Maybe<>` allows both undefined and null. I'd rather be too cautious here and then later on change Maybe type where possible - it means reworking types throughout the whole app, so I don't think we should address it right now ;)

Reply:
I actually just digged further, and the main issue here is that on non-hourly aggregrates we don't get capacity data from the backend. 
So therefore capacity values can be either `undefined`, `null` or a `number`. 
We could write it in either of the ways below, but don't know if it matters much :)
```
capacity?: number | null; 
// capacity: Maybe<number>;
```

Reply:
Maybe we should start returning data for these even if it's null in preparation for capacity parsers (some zones should already have de facto capacity parsers as it's returned as a part of the production object but I don't know how this is saved)?

ENTSO-E TP already have yearly installed capacity for most zones so we just need to save this to a database really. (and use the capacity in the config as a fallback.)

Reply:
But then we're back to the problem of trying to reduce backend payload size! :D

Reply:
I don't think it's the zone state files that are the issue right now but the global one (that is always downloaded and refetched).